src/problem.ts

```diff
@@ -1,62 +1,77 @@
-import { httpObject, ProblemInterface } from "types";
+import { DEFAULT_KEYS } from "./constants";
+import {
+  HttpObject,
+  ProblemInterface,
+  ToJsonParamType,
+  UpdateProblemParamType,
+} from "./types";
 
-enum COPY_MODE {
-    SKIP_PROPS = 'skipProps',
-    LEAVE_PROPS = 'leaveProps'
-}
+import { COPY_MODE } from "./constants";
 
 export class Problem extends Error implements ProblemInterface {
-    public type: string;
-    public title: string;
-    public instance?: string;
-    public detail?: string;
-    public http?: httpObject
-    [key: string]: any;
-
-
-    constructor(problem: ProblemInterface, customKeys?: string[]) {
-        super(problem.detail || problem.title);
-        this.http = problem.http
-        this.type = problem.type
-        this.title = problem.title;
-        this.detail = problem.detail;
-        this.instance = problem.instance;
-        this.stack = problem.stack;
-        customKeys?.map((customKey) => {
-            this[customKey] = problem[customKey];
-        })
-    }
+  public type: string;
+  public title: string;
+  public instance?: string;
+  public detail?: string;
+  public http?: HttpObject;
+  [key: string]: any;
 
-   copy(problem: ProblemInterface, mode: COPY_MODE, props: string[]): Problem {
-        switch (mode) {
-
-            case COPY_MODE.LEAVE_PROPS:
-                return new Problem(problem, props);
-
-            case COPY_MODE.SKIP_PROPS:
-            default:
-                let keysToBeCopied: string[] = [];
-                for (let key in problem) {
-                    if (props.includes(key))
-                        continue;
-                    keysToBeCopied.push(key)
-                }
-                return new Problem(problem, keysToBeCopied)
-        }
-    };
+  constructor(protected readonly problem: ProblemInterface) {
+    super(problem.detail || problem.title);
+    this.http = problem.http;
+    this.type = problem.type;
+    this.title = problem.title;
+    this.detail = problem.detail;
+    this.instance = problem.instance;
+    this.stack = problem.stack;
 
-   toJSON(problem: Problem, includeStack = false): ProblemInterface {
+    // add extra keys
+    Object.keys(problem)
+      .filter((el) => !DEFAULT_KEYS.includes(el))
+      .forEach((k) => (this[k] = problem[k]));
+  }
 
-        const { name, message, stack, ...rest } = problem;
+  copy(mode: COPY_MODE = COPY_MODE.LEAVE_PROPS, props: string[] = []): Problem {
+    switch (mode) {
+      case COPY_MODE.LEAVE_PROPS:
+        const newProblem = new Problem(this.problem);
+        props.forEach((prop) => (newProblem[prop] = undefined));
```

Comment:
If you leave some props, you should set undefined to props these ones which you don't wanna leave.

Reply:
Thanks for the clarification, modifying it right away